### ISSUE TITLE
Quarks on Windows 10

### ISSUE THREAD

**Issue by d0kt0r0** [supercollider/supercollider#2196]
With a freshly downloaded 3.7.2 64-bit build for Windows, on a fairly basic, clean, new Windows 10 installation, with the common Git for Windows binary installed, various Quarks operations that rely on git fail. An immediate workaround is to add the path to git to the system's PATH. This is however not a particularly friendly thing to ask users to do, and undermines what could otherwise be a very friendly experience with the Quarks system. 

It would be great if SC builds on Windows would work with this common Git installation method, out of the box, without requiring (for example: new) users to meddle with their system's PATH.

For the benefit of any users wishing to work around this on Windows, for the meantime, here's how I did it:
-Click on Settings
-then System
-then About
-then System Info (at the very bottom)
-then Advanced System Settings
-then Environment Variables
-then Edit with the focus on PATH
-then click New and enter C:/Program Files/Git/bin
-close and relaunch SuperCollider (just rebooting interpreter was _not_ enough)
-confirm that evaluating Quarks.gui now hopefully gives you a list of available Quarks

**Comment by crucialfelix** [comment]
Was this what you installed: https://git-for-windows.github.io/ ? 

Does it really not install git to the path ? 

How then are you supposed to use it, how do people normally install git and then go into their terminal app and run git ?

That's the basic requirement we need for Quarks to work: git has to be installed and it has to be on your path. If it isn't on your path, then it kind of isn't really installed.

We could add classvar to the Git class to let you set the path to git.  But I still expect that practically everybody who installs git by any means should have it on their path.

**Comment by d0kt0r0** [comment]
Yes that should be the one.


It does come with its own MSYS shell where it’s added to the path.


I don’t know why it isn’t added to the general path by the git for windows installer, so I guess there may be an issue here for that project as well.


In the meantime, a little “on Windows check for C:\Program Files\Git\bin\git.exe if other things have failed before giving up” might go a long way to making the experience smoother for users with less experience configuring things.  I’d vouch
 for that in addition to the ability you mention to set the path - because the latter still assumes that the user understands something about the PATH - whereas the rest of the Quark system seems to aim at being more friendly to users less familiar with such
 things.





On Jun 20, 2016, at 4:59 PM, Chris Sattinger <notifications@github.com> wrote:


Was this what you installed: 
https://git-for-windows.github.io/ ? 
Does it really not install git to the path ? 
How then are you supposed to use it, how do people normally install git and then go into their terminal app and run git ?
That's the basic requirement we need for Quarks to work: git has to be installed and it has to be on your path. If it isn't on your path, then it kind of isn't really installed.
We could add classvar to the Git class to let you set the path to git. But I still expect that practically everybody who installs git by any means should have it on their path.
—
You are receiving this because you authored the thread.
Reply to this email directly, 
view it on GitHub, or 
mute the thread.

**Comment by crucialfelix** [comment]
I agree it should be simple. 

Though we do make the assumption that git is something that everybody would have or should have. Not just for supercollider.

You do have it - but that installer comes with its own shell. hmm. That's strange. 

It already is checking for git.  If you run this in supercollider:

    Git.checkForGit

you might see a "Quarks requires git to be installed" error.  This check could be improved to actually set the path automatically if it finds it. We could try to find it and guess in some locations if "where git" fails.

What does this do in your terminal:

    where git

it should locate git, but only if it is on your path already.

This is the other popular git for windows: https://git-scm.com/download/win

How is the new unix shell inside windows working ? Ideally we should switch to that for Windows 10.

**Comment by jamshark70** [comment]
> but that installer comes with its own shell. hmm. That's strange.

A likely explanation (speculation here) is that they tried to make it work with the msdos shell but failed because msdos is not close enough to UNIX.

**Comment by bagong** [comment]
No, it also works with the normal Windows shell, the bash shell is just a different mode that offers a few more features. Adding git to the path is part of the install procedure on Windows, the git installer offers it as default (but a user could reject it). (@crucialfelix, in Windows you add multiple application-"paths" to a path environment variable rather than install a program into the path (in Windows it's considered bad style to install into the system folders))
I didn't spot an environment variable for git. Also there are 32- and 64-bit installers now which install to different folders, and of course a user could choose any path. So I don't see something robust that could be done without git already being added to the path. I guess we have to rely on the documentation here.
